### PR TITLE
Generate AI release notes for monorepo members

### LINKED ISSUE #83: Release notes section is empty for monorepo members
## Context

Monorepo-mode release PRs land with an empty `## Release Notes` block. PR #80 ("Release code-review-action 0.1.0") body shows:

```
## 0.1.0 (2026-05-28)

## Release Notes


📋 [Detailed changelog](...)
```

In standalone mode, `.github/actions/release-action/action.yml` has a dedicated `Generate Release Notes` step that calls `src/generateReleaseNotes.ts`, hitting the Anthropic Messages API with the changelog and writing the summary to `.release_bot/release_notes.md` for the PR body assembly to pick up. That step is gated on `steps.detect-type.outputs.mode == 'standalone'` — it is never executed in monorepo mode.

The monorepo path runs entirely in TypeScript via `src/run.ts` → `src/monorepo/runCreate.ts`. That code calls `generateChangelog` and writes the raw changelog to `<member>/.release_notes/<version>.md` (`runCreate.ts:213`), but never invokes `generateReleaseNotes()`. So the AI summary is silently missing for every monorepo member, even when `ANTHROPIC_API_KEY` is set on the action (it is — we pass `secrets.ANTHROPIC_RELEASE` in our workflow).

Related: #67 (closed)

## What

Wire `generateReleaseNotes()` into the monorepo create flow so per-member release notes get the AI summary, identical in shape to the standalone path. The visible outcome: PRs like #80 carry a populated `## Release Notes` section instead of a blank one.

## Why

The AI summary is the single most reader-friendly part of a release PR — it turns a raw changelog into a paragraph a human can skim before approving. Without it, reviewers must read every commit subject and infer impact themselves, which removes most of the value the release pipeline was meant to add (#67). Every monorepo member ships with this gap today.

## Scope

- **In scope:**
  - Invoke `generateReleaseNotes()` per member inside `src/monorepo/runCreate.ts`, scoped to the member's path so the changelog input and the output file are member-local
  - Reuse the existing `ANTHROPIC_API_KEY` env (already plumbed via `action.yml`'s `Run monorepo release` step)
  - Match the standalone fallback behaviour: on API failure, fall back to the full changelog (per `generateReleaseNotes.ts:90`) rather than leaving the section empty
- **Out of scope:**
  - Refactoring `generateReleaseNotes.ts` itself — it is already path-aware via env or argument
  - Changing the standalone path
  - Adding new model selection or prompt knobs

## Solution

In `src/monorepo/runCreate.ts`, after `generateChangelog` produces the per-member `release` body and it is written to `<member>/.release_notes/<version>.md`, call `generateReleaseNotes()` with the member path as cwd (or pass paths explicitly to the helper). The helper already handles missing API key and API errors with `console.log("::warning::...")` plus fallback to the full changelog, so the wiring is purely additive: read the changelog, summarize, overwrite the notes file with the enriched content.

```text
generateChangelog(member) ──▶ raw changelog
                  │
                  ▼
   write <member>/.release_notes/<version>.md  (current behaviour)
                  │
                  ▼  (new)
   generateReleaseNotes(member)
                  │
                  ├─▶ ANTHROPIC_API_KEY missing → keep raw changelog
                  ├─▶ API error                → log warning, keep raw
                  └─▶ success                  → overwrite with AI summary
```

Verify by running `release.yml` via `workflow_dispatch` after the fix and inspecting any one of the resulting release PRs — the `## Release Notes` block should contain a human-readable paragraph, not blank lines.

### PR DESCRIPTION
**Pull request by awinogradov**
Wires the AI release-note generation into the monorepo create path so per-member release PRs ship with a populated `## Release Notes` block instead of blank space. Until now `runCreate.ts` only wrote the raw changelog to `<member>/.release_notes/<version>.md`; the `Generate Release Notes` action.yml step was gated on `mode == 'standalone'` and never ran for monorepo members.

- Extract `runReleaseNotes(cwd)` from `generateReleaseNotes.ts`'s `main()` so the same orchestration (API call + verify-with-fallback) can be reused from anywhere
- Parameterize `generateWithApi` with `cwd` so `tickets.json`, `pr_descriptions.yml`, and `serviceContext` resolve under the member directory in monorepo mode instead of always the repo root
- Call `runReleaseNotes(member.path)` and `insertReleaseNotes({ cwd: member.path, version })` inside `emitMemberArtifacts` after `.release_bot/body` is written, mirroring the standalone `Generate Release Notes` + `Update Release Notes File` action.yml step sequence
- `runReleaseNotes` falls back to the full changelog when `ANTHROPIC_API_KEY` is missing or the API call fails, so the existing test fixtures that omit the key still emit non-empty notes

---

**Release notes:**

- Monorepo release PRs now include the AI-summarized release notes for every member, matching standalone-mode behaviour

---

**Issues:**

Closes #83

### CODE REVIEW

File: .github/actions/release-action/src/generateReleaseNotes.ts [hunk 1]
```diff
@@ -102,17 +104,22 @@ export async function verifyReleaseNotes(notesPath: string, bodyPath: string): P
   await Bun.write(notesPath, "- See changelog for detailed changes\n");
 }
 
-async function generateWithApi(apiKey: string, notesPath: string, bodyPath: string): Promise<void> {
+export async function generateWithApi(
```
[the remark this answers is not in the document]
Comment (awinogradov) [reply .github/actions/release-action/src/generateReleaseNotes.ts:107]: @symbiot-bot Fixed — added a runReleaseNotes test that writes `.release_bot/body` under a tempdir and asserts the AI summary lands at `<cwd>/.release_bot/release_notes.md` (not `process.cwd()`). `generateWithApi`'s cwd-relative reads of tickets.json / pr_descriptions.yml / serviceContext are exercised through that path; a direct catch-branch test on `generateWithApi` with a missing body file covers the failure case.

File: .github/actions/release-action/src/generateReleaseNotes.ts [hunk 2]
```diff
@@ -130,18 +137,31 @@ async function generateWithApi(apiKey: string, notesPath: string, bodyPath: stri
...
+/**
+ * Generate and verify release notes for a given working directory. Reads the
+ * full changelog from `<cwd>/.release_bot/body`, writes the AI summary to
+ * `<cwd>/.release_bot/release_notes.md`, and falls back to the full changelog
+ * when the API key is missing or the call fails.
+ *
+ * Used by the standalone `main()` (cwd = repo root) and by the monorepo
+ * `emitMemberArtifacts` (cwd = member path).
+ */
+export async function runReleaseNotes(cwd = process.cwd()): Promise<void> {
```
[the remark this answers is not in the document]
Comment (awinogradov) [reply .github/actions/release-action/src/generateReleaseNotes.ts]: @symbiot-bot Fixed — new `runReleaseNotes` describe block has two tests: one verifies the AI summary file lands under the supplied cwd, the other asserts `process.cwd()/.release_bot/release_notes.md` is NOT created when a non-default cwd is supplied, locking in the monorepo isolation guarantee.

File: .github/actions/release-action/src/generateReleaseNotes.ts [hunk 3]
```diff
@@ -130,18 +137,31 @@ async function generateWithApi(apiKey: string, notesPath: string, bodyPath: stri
...
 
   if (apiKey) {
-    await generateWithApi(apiKey, notesPath, bodyPath);
+    await generateWithApi(apiKey, notesPath, bodyPath, cwd);
   }
 
   await verifyReleaseNotes(notesPath, bodyPath);
 }
 
+async function main(): Promise<void> {
```
[the remark this answers is not in the document]
Comment (awinogradov) [reply .github/actions/release-action/src/generateReleaseNotes.ts]: @symbiot-bot Fixed — dropped the `main()` wrapper and call `runReleaseNotes()` directly from the `import.meta.main` guard.

File: .github/actions/release-action/src/generateReleaseNotes.test.ts [hunk 4]
```diff
@@ -369,3 +371,56 @@ describe("callAnthropicApi", () => {
     ).rejects.toThrow("Anthropic API returned no text content");
   });
 });
+
+describe("runReleaseNotes", () => {
+  const originalApiKey = process.env.ANTHROPIC_API_KEY;
```
[the remark this answers is not in the document]
Comment (awinogradov) [reply .github/actions/release-action/src/generateReleaseNotes.test.ts]: @symbiot-bot Fixed — replaced the describe-scope capture with `beforeEach` / `afterEach` hooks that snapshot `ANTHROPIC_API_KEY` before each test and restore (or unset) it after, so prior-test env mutations can't corrupt the saved value. Same pattern applied to the `generateWithApi` describe block.

File: .github/actions/release-action/src/generateReleaseNotes.test.ts [hunk 5]
```diff
@@ -369,3 +371,56 @@ describe("callAnthropicApi", () => {
     ).rejects.toThrow("Anthropic API returned no text content");
   });
 });
+
+describe("runReleaseNotes", () => {
```
[the remark this answers is not in the document]
Comment (awinogradov) [reply .github/actions/release-action/src/generateReleaseNotes.test.ts:375]: @symbiot-bot Fixed — threaded an optional `AnthropicMessages` mock through `runReleaseNotes` and `generateWithApi` (mirroring `callAnthropicApi`'s existing testing seam). New `runReleaseNotes` test sets `ANTHROPIC_API_KEY`, passes a mock client, and asserts the AI summary lands at `<cwd>/.release_bot/release_notes.md` — exercising the previously-uncovered `if (apiKey)` branch.

File: .github/actions/release-action/src/generateReleaseNotes.test.ts [hunk 6]
```diff
@@ -369,3 +371,56 @@ describe("callAnthropicApi", () => {
...
+        const cwdNotes = Bun.file(join(process.cwd(), ".release_bot/release_notes.md"));
+        expect(await cwdNotes.exists()).toBe(false);
+      } finally {
+        if (originalApiKey !== undefined) process.env.ANTHROPIC_API_KEY = originalApiKey;
+      }
+    }));
+});
+
+describe("generateWithApi", () => {
+  test("swallows errors and reads paths relative to the supplied cwd", () =>
```
[the remark this answers is not in the document]
Comment (awinogradov) [reply .github/actions/release-action/src/generateReleaseNotes.test.ts:446]: @symbiot-bot Fixed — added two `generateWithApi` tests covering the happy path: one writes a body, supplies a mock returning "Summary", and asserts the notes file contains it; the other places `tickets.json` and `pr_descriptions.yml` under cwd, mocks the client, and asserts the captured user message contains both — proving cwd-relative resolution AND the body file write path.

**Comment by awinogradov** [comment]
@symbiot-bot Fixed — added `src/insert-release-notes.test.ts` with five tests covering the happy path (splicing notes into `CHANGELOG.md` and `.release_notes/<v>.md` under the version header) plus the missing-source / empty-source / absent-version-header / prepend-on-no-header branches.
